sentry-android-core/src/main/java/io/sentry/android/core/util/ConnectivityChecker.java

```diff
@@ -36,43 +45,49 @@ private ConnectivityChecker() {}
     // connectivityManager.registerDefaultNetworkCallback(...)
   }
 
+  /**
+   * Returns whether its connected or not
+   *
+   * @param context the Context
+   * @param connectivityManager the ConnectivityManager
+   * @param logger the Logger
+   * @return true if connected or no permission to check, false otherwise
+   */
   @SuppressWarnings({"deprecation", "MissingPermission"})
-  private static Boolean isConnected(
-      Context context, ConnectivityManager connectivityManager, ILogger logger) {
-    android.net.NetworkInfo activeNetwork =
-        getActiveNetworkInfo(context, connectivityManager, logger);
-
-    if (activeNetwork == null) {
-      logger.log(SentryLevel.INFO, "NetworkInfo is null and cannot check network status");
-      return null;
-    }
-    return activeNetwork.isConnected();
-  }
-
-  @SuppressWarnings({"deprecation", "MissingPermission"})
-  private static @Nullable android.net.NetworkInfo getActiveNetworkInfo(
-      @NotNull Context context,
-      @NotNull ConnectivityManager connectivityManager,
-      @NotNull ILogger logger) {
+  private static @NotNull ConnectionStatus isConnected(
```

Comment:
`l`: Now that we have the enum as return type maybe we should rename this to `getConnectionStatus` or something similar?